### ISSUE TITLE
Do not attempt to install kvm on self-hosted runners

### ISSUE THREAD

**Issue by dpc** [DeterminateSystems/nix-installer-action#132]
I have a CI that have some things working on self-hosted runners, and some on normal Github runners.

Currently, after switching:

```
-      - uses: cachix/install-nix-action@v30
+      - uses: DeterminateSystems/nix-installer-action@v16
```

The CI fails on self-hosted runners, because it tries to install KVM, but there is no sudo.

It seems to me that if the Nix was already installed, the machine is probably a self-hosted one, and there's no point trying to install KVM.

I will probably end up disabling KVM, but ideally I would be happy to have it enabled on non-self-hosted machines.

**Edit:**

Seems like currently the existing Nix installation is not even correctly detected.

self-hosted runners just have `nix` in the PATH, and do not actually require installing anything. The action is still useful if it needs to run on a Github Actions standard runner in a VM.

**Comment by dpc** [comment]
[Ehh... I forgot this doesn't work](https://github.com/fedimint/fedimint/actions/runs/13194597650/job/36833479852). Bummer. I really would love to switch.

**Comment by grahamc** [comment]
Hmm... interesting. So, installing Nix requires sudo. @dpc does your machine already have Nix? I have a fix for the KVM + sudo issue, but obviously it won't help if you don't have Nix installed.

**Comment by dpc** [comment]
> Hmm... interesting. So, installing Nix requires sudo. [@dpc](https://github.com/dpc) does your machine already have Nix? I have a fix for the KVM + sudo issue, but obviously it won't help if you don't have Nix installed.

Yes. It's a self-hosted runner. A hetzner box with NixOS installed and github runners configured.

It does require some working around actions sometimes, because some github actions assume that they are running on a fresh VM Ubuntu, but mostly works.

I wouldn't really need `nix-installer-action` in the first place there *BUT* the same workflows are used when using MacOS builders, and possibly in the VM when people fork our repo. So it would be really nice if DeterminateSystems/nix-installer-action could just detect that Nix is installed and `sudo` doesn't work and just assumed it's OK instead of failing.
